### PR TITLE
ZSink#raceBoth: relax the type bounds of the return type

### PR DESCRIPTION
**Pull request by sideeffffect**

### CODE REVIEW

File: core/shared/src/main/scala/zio/FiberId.scala [hunk 1]
```diff
@@ -76,7 +76,7 @@ object FiberId {
 
   private[zio] val _fiberCounter = new java.util.concurrent.atomic.AtomicInteger(0)
 
-  case object None                                           extends FiberId
+  final case object None                                     extends FiberId
```
Comment (adamgfraser) [inline core/shared/src/main/scala/zio/FiberId.scala:79]: I think this will result in an error on Scala 3.

**Review by adamgfraser — APPROVED** [review]
Good to go when CI is passing.

File: core/shared/src/main/scala/zio/FiberId.scala (hunk 1 above)
Comment (sideeffffect) [reply core/shared/src/main/scala/zio/FiberId.scala:79]: Seems like it passes :crossed_fingers:

**Comment by sideeffffect** [comment]
All green :tada:

**Pull request merged**